### ISSUE TITLE
does subset() also subset the flags?

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1410]

**Comment by dankelley** [comment]
No, it doesn't, as shown below. I'll fix.
```R
library(oce)
data(section)
stn <- section[["station", 100]]
stnTopKm <- subset(stn, pressure < 1000)
expect_equal(length(stn[["salinity"]]), length(stn[["salinityFlag"]]))
expect_equal(length(stnTopKm[["salinity"]]), length(stnTopKm[["salinityFlag"]]))
```

**Comment by dankelley** [comment]
Hm. This is class-dependent, so I'll need to have a checklist:
* [ ] AllClass.R:264:setMethod(f="subset",
* [ ] adp.R:668:setMethod(f="subset",
* [ ] adv.R:271:setMethod(f="subset",
* [ ] amsr.R:283:setMethod(f="subset",
* [ ] argo.R:401:setMethod(f="subset",
* [ ] cm.R:161:setMethod(f="subset",
* [ ] coastline.R:108:setMethod(f="subset",
* [ ] ctd.R:3558:setMethod(f="subset",
* [ ] echosounder.R:234:setMethod(f="subset",
* [ ] lobo.R:161:setMethod(f="subset",
* [ ] met.R:137:setMethod(f="subset",
* [ ] odf.R:92:setMethod(f="subset",
* [ ] rsk.R:169:setMethod(f="subset",
* [ ] sealevel.R:169:setMethod(f="subset",
* [ ] section.R:547:setMethod(f="subset",
* [ ] topo.R:163:setMethod(f="subset",

**Comment by dankelley** [comment]
Fixed in "develop" branch commit b1b432721d766d5342c2f1e8e91851667b9264b6, and `tests/testthat/test_flags` now has specific tests for `ctd`, `odf` and `adp` classes.
